### PR TITLE
The argentinian location fields should appear in the checkout only if…

### PR DESCRIPTION
**Pull request by maq-adhoc**
The Argentinian location fields should only be available in the website checkout if the ordering company is located in Argentina. 

If I use a cart retrieval link with a sales order from a company located in another country the partner should not display or validate the AFIP related information.

This pr modifies this validation by replacing webiste company by order company. 

Current behavior before PR: 
I always show the location fields on a website if it is related to an argentine company,

Desired behavior after PR is merged:

I only show the location fields if the sales order is related to an Argentinian company.

### CODE REVIEW

**Review by niyasraphy — COMMENTED** [review]
hi @maq-adhoc ,
am not sure about the functionality, but could you sign the cla and improve the commit message as per guidelines.

guidelines: https://www.odoo.com/documentation/16.0/contributing/development/git_guidelines.html
cla:  https://github.com/odoo/odoo/blob/16.0/doc/cla/sign-cla.md

**Comment by qdp-odoo** [comment]
@maq-adhoc code looks ok to me, but it's more in the scope of website or sale ... 

that being said, the commit message is still not compliant with our guidelines. Can you rephrase, please?

> [FIX] l10n_ar_website_sale: visibilty of argentinian fields
> The argentinian location fields should appear in the checkout only if the sales order belongs to an argentinian company. E.g. a recovery cart link from a company in another location should not show the argentina fields even if the website is located in argentina.
